sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -441,7 +441,15 @@ class Analyzer(
   object ResolveRelations extends Rule[LogicalPlan] {
     private def lookupTableFromCatalog(u: UnresolvedRelation): LogicalPlan = {
       try {
-        catalog.lookupRelation(u.tableIdentifier, u.alias)
+        val table = catalog.lookupRelation(u.tableIdentifier, u.alias)
+        table match {
+          case SubqueryAlias(_, s: SimpleCatalogRelation)
```

Comment:
I'm not sure if this is useful, `SimpleCatalogRelation` is always unresolved and we can provide better message at `CheckAnalysis` if we want to.